### DISCUSSION TITLE
Anisotropic material & composites laminate modeling

### DISCUSSION THREAD

**Discussion by yvanblanchard** [adtzlr/felupe#804] [Q&A]
Hello,

I would like to know if aniostropic material & composites laminate (layers) can be modeled, using fiber direction field ? (i.e. CFRP)

Thank you !

**Comment by adtzlr** [comment]
Hi, we have done that with constant fiber directions in https://github.com/adtzlr/fiberreinforcedrubber using [matadi](https://github.com/adtzlr/matadi).

A fiber direction field should be possible in FElupe, just provide the direction field with shape `(3, nq, nc)`, where the first axis are the vector components, the second axis contains the quadrature points per cell (=element) and the last axis the cells.  You'd have to define your strain energy function manually because there is no built-in formulation available. See [Hyperelastic](https://felupe.readthedocs.io/en/latest/felupe/constitution/hyperelasticity.html#felupe.Hyperelastic) for more details.

Do you have any specific strain energy density formulation in mind?

We have no layers, shells, rebars or whatever these special element formulations are called in the different codes. FElupe has just solids built in.
